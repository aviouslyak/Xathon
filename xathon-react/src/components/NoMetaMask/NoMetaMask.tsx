import React from "react";

const NoMetaMask: React.FC = () => {
  return (
    <div
      data-testid="no-meta-mask wrapper"
      className="w-screen flex flex-col items-center mt-5"
    >
      <h3 className="text-2xl">
        MetaMask is not installed. It is required for this application to
        function correctly.
      </h3>
      <h5 className="text-xl">
        Get it&nbsp;
        <a
          target="_blank"
          className="text-blue-500 underline"
          rel="noopener noreferrer"
          href="https://metamask.io"
        >
          here
        </a>
        &nbsp;and refresh the page.
      </h5>
    </div>
  );
};

export default NoMetaMask;
