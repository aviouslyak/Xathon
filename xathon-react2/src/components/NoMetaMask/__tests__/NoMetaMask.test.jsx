import { render, screen } from "@testing-library/react";
import NoMetaMask from "../NoMetaMask";

describe("<NoMetaMask />", () => {
  test("matches snapshot", () => {
    render(<NoMetaMask />);
    const component = screen.getByTestId("no-meta-mask wrapper");
    expect(component).toMatchSnapshot();
  });
});
