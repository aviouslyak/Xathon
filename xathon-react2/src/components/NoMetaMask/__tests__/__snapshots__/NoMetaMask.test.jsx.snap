// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMetaMask /> matches snapshot 1`] = `
<div
  class="w-screen flex flex-col items-center mt-5"
  data-testid="no-meta-mask wrapper"
>
  <h3
    class="text-2xl"
  >
    MetaMask is not installed. It is required for this application to function correctly.
  </h3>
  <h5
    class="text-xl"
  >
    Get it 
    <a
      class="text-blue-500 underline"
      href="https://metamask.io"
      rel="noopener noreferrer"
      target="_blank"
    >
      here
    </a>
     and refresh the page.
  </h5>
</div>
`;
